src/webview/manage-repository/app/showRepositories.tsx

```diff
@@ -0,0 +1,250 @@
+            newRepoInput: {
+                name: '',
+                error: false,
+                helpText: ''
+            }
+        }
+    }
+
+    componentDidMount(): void {
+        VSCodeMessage.onMessage((message) => {
+            if (message.data.action === 'getRepositoryList') {
+                this.setState({
+                    repositories: message.data.repositories
+                });
+            } else if (message.data.action === 'validateNewName') {
+                this.setState({
+                    newRepoInput: {
+                        name: message.data.name,
+                        error: message.data.error,
+                        helpText: message.data.helpText
+                    }
+                })
+            }
+        });
+    }
+
+    handleDialog = (repoName: string, isEdit = true): void => {
+        this.setState({
+            openedRepo: repoName,
+            newRepoInput: {
+                name: repoName,
+                error: false,
+                helpText: ''
+            },
+            openDeleteDialog: isEdit ? false : !this.state.openDeleteDialog,
+            openEditDialog: isEdit ? !this.state.openEditDialog : false
+        });
+    }
+
+    delete = (repo: string): void => {
+        this.CloseDialog();
+        VSCodeMessage.postMessage({
+            action: `deleteRepo`,
+            name: repo
+        });
+    }
+
+    CloseDialog = (): void => {
+        this.setState({
+            openEditDialog: false,
+            openDeleteDialog: false,
+            openedRepo: '',
+            newRepoInput: {
+                name: '',
+                error: false,
+                helpText: ''
+            }
+        });
+    }
+
+    setRepoName = (value: string): boolean => {
+        this.setState({
+            openEditDialog: true,
+            newRepoInput: {
+                name: value,
+                error: false,
+                helpText: ''
+            }
+        });
+        return this.state.openEditDialog;
+    }
+
+    validateName = (value: string): void => {
+        VSCodeMessage.postMessage({
+            action: `validateNewName`,
+            name: value
+        });
+    }
+
+    handleDisable = (): boolean => {
+        return this.state.newRepoInput.name.length === 0 || this.state.newRepoInput.error
+    }
+
+    rename(oldName: string, newName: string): void {
+        this.CloseDialog();
+        VSCodeMessage.postMessage({
+            action: `renameRepo`,
+            oldName: oldName,
+            newName: newName
+        });
+    }
+
+    render(): React.ReactNode {
+        const { newRepoInput, openedRepo, openDeleteDialog, openEditDialog, repositories } = this.state;
+        return (
+            <Grid container spacing={{ xs: 2, md: 3 }} columns={{ xs: 4, sm: 8, md: 12 }}>
+                {
+                    repositories.map((repoName: string, index: number) => (
+                        <Grid item xs={2} sm={4} md={4} key={index}>
```

Comment:
@msivasubramaniaan If there are multiple repository (say more than 5), it will take to much space in the UI to show them and it will be cumbersome.

We should use React Table to show the information of the Repository available, with Edit and Delete option included. This way we can scale up the records as added.

Reply:
Reference example: https://codesandbox.io/p/sandbox/github/KevinVandy/material-react-table/tree/main/apps/material-react-table-docs/examples/editing-crud/sandbox?file=/src/TS.tsx:1,1